PROTO_tests/tests/series_test.py

```diff
@@ -0,0 +1,193 @@
+            s = ak.Series(data=val, index=mi)
+            lk = s.locate(mi[0])
+            assert isinstance(lk, ak.Series)
+            assert lk.index.index == mi[0].index
+            assert lk.values[0] == val[0]
+
+            # ensure error with scalar and multi-index
+            with pytest.raises(TypeError):
+                s.locate(0)
+
+            with pytest.raises(TypeError):
+                s.locate([0, 2])
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    @pytest.mark.parametrize("dtype", NO_STRING)
+    def test_add(self, prob_size, dtype):
+        added = ak.Series(ak.arange(prob_size // 2, dtype=dtype)).add(
+            ak.Series(
+                data=ak.arange(prob_size // 2, prob_size, dtype=dtype),
+                index=ak.arange(prob_size // 2, prob_size),
+            )
+        )
+        assert added.index.to_list() == ak.arange(prob_size).to_list()
+        assert added.values.to_list() == ak.arange(prob_size, dtype=dtype).to_list()
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    @pytest.mark.parametrize("dtype", [ak.int64, ak.uint64, ak.float64])
+    def test_topn(self, prob_size, dtype):
+        prob_size_half = prob_size // 2
+        top = ak.Series(ak.arange(prob_size, dtype=dtype)).topn(prob_size_half)
+        assert top.values.to_list() == list(range(prob_size - 1, prob_size_half - 1, -1))
+        assert top.index.to_list() == list(range(prob_size - 1, prob_size_half - 1, -1))
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    @pytest.mark.parametrize("dtype", NUMERICAL_TYPES)
+    @pytest.mark.parametrize("dtype_index", NUMERICAL_TYPES)
+    def test_sort(self, prob_size, dtype, dtype_index):
+        def gen_perm(n):
+            idx_left = set(range(n))
+            perm = np.arange(n)
+            while len(idx_left) >= 2:
+                inds = np.random.choice(list(idx_left), 2, replace=False)
+                tmp = perm[inds[0]]
+                perm[inds[0]] = perm[inds[1]]
+                perm[inds[1]] = tmp
+                idx_left.remove(inds[0])
+                idx_left.remove(inds[1])
+            return perm
+
+        ordered = ak.arange(prob_size, dtype=dtype)
+        perm = ak.array(gen_perm(prob_size), dtype=dtype_index)
+
+        idx_sort = ak.Series(data=ordered, index=perm).sort_index()
+        assert idx_sort.index.to_list() == ordered.to_list()
+        assert idx_sort.values.to_list() == perm.to_list()
+
+        val_sort = ak.Series(data=perm, index=ordered).sort_values()
+        assert val_sort.index.to_pandas().tolist() == perm.to_list()
+        assert val_sort.values.to_list() == ordered.to_list()
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    @pytest.mark.parametrize("dtype", DTYPES)
+    def test_head_tail(self, prob_size, dtype):
+        s = ak.Series(ak.arange(prob_size, dtype=dtype))
+
+        for i in range(prob_size):
+            head = s.head(i)
```

Comment:
this is should not be parmeterized on problem size. This will take forever for even a moderate problem prob size (say 1000) because we are doing thousands of arkouda operations and array creations, not even mentioning doing `list(range(i))` for every i less than prob_size. I wouldn't do this for any more than `n=10`

Reply:
Updated